src/components/project/MonitoringTab.js

```diff
@@ -57,22 +61,35 @@ class MonitoringTab extends React.Component {
         os: snapshot.val()
       })
     })
+    this.updateWindowDimensions()
+    window.addEventListener('resize', this.updateWindowDimensions)
   }
 
   componentWillUnmount() {
     this.heartbeatRef.off()
     this.httpRef.off()
     this.osRef.off()
+    window.removeEventListener('resize', this.updateWindowDimensions)
```

Comment:
I think it might be better if you isolate the resizing logic to each component rather than making it tightly coupled with the MonitoringTab component.

Reply:
For simplicity, I think it should be left in MonitoringTab. Ideally the components should be resizing themselves, but since this fix is required it is convenient to keep the messy fixed measuring logic in one place. Again I wish these components didn't require their size to be specifically defined at all.